spectral_cube/lower_dimensional_structures.py

```diff
@@ -68,6 +72,52 @@ def write(self, filename, format=None, overwrite=False):
             raise ValueError("Unknown format '{0}' - the only available "
                              "format at this time is 'fits'")
 
+    def to(self, unit, equivalencies=[]):
+        """
+        Return a new ``LowerDimensionalObject'' of the same class with the
+        specified unit.  See `astropy.units.Quantity.to` for further details.
+        """
+        converted_array = u.Quantity.to(self, unit,
+                                        equivalencies=equivalencies).value
+
+        # use private versions of variables, not the generated property
+        # versions
+        # Not entirely sure the use of __class__ here is kosher, but we do want
+        # self.__class__, not super()
+        new = self.__class__(value=converted_array, unit=unit, copy=True,
+                             wcs=self._wcs, meta=self._meta, mask=self._mask,
+                             header=self._header)
+
+        return new
+
+    def __getitem__(self, key):
+        """
+        Return a new ``LowerDimensionalObject'' of the same class while keeping
+        other properties fixed.
+        """
+        new_qty = super(LowerDimensionalObject, self).__getitem__(key)
+
+        if new_qty.ndim < 2:
+            # do not return a projection
+            return u.Quantity(new_qty)
```

Comment:
There might be.  I chose to ignore it for now because it's not trivial, but I think we could inspect the WCS and see if one of the axes is spectral; if so, we'd then return the `OneDSpectrum`